Model/AdyenPaypalUpdateOrder.php

```diff
@@ -85,17 +85,36 @@ public function __construct(
     /**
      * @param int $adyenCartId
      * @param string $paymentData
-     * @param array $deliveryMethods
+     * @param string $deliveryMethods
      * @return string
      * @throws NoSuchEntityException
      * @throws ValidatorException
      */
-    public function execute(int $adyenCartId, string $paymentData, $deliveryMethods = []): string
+    public function execute(int $adyenCartId, string $paymentData, string $deliveryMethods = ''): string
     {
         /** @var Quote $quote */
         $quote = $this->cartRepository->get($adyenCartId);
         $merchantReference = $quote->getReservedOrderId();
+        $deliveryMethods = json_decode($deliveryMethods, true);
 
+        foreach ($deliveryMethods as &$method) {
+            // Ensure the amount value is an integer
+            $method['amount']['value'] = (int) $method['amount']['value'];
+
+            // Validate the current method
+            $validatedMethod = $this->deliveryMethodValidator->getValidatedDeliveryMethods([$method]);
```

Comment:
`getValidatedDeliveryMethods()` method accepts and returns `array`. Instead of passing a single item in brackets and obtaining the value with first index, this method can be called before `foreach` with `$deliveryMethods` variable.